addons/sale_timesheet/tests/test_sale_timesheet.py

```diff
@@ -969,6 +969,43 @@ def test_onchange_uom_service_product(self):
                 product_form.detailed_type = 'consu'
                 self.assertEqual(product_form.uom_id.id, uom_kg.id)
 
+    def test_invoice_creation_running_timer(self):
+        """ Test that the creation of invoice stop and take into account the running timers. """
+        self.env.user.employee_id = self.env['hr.employee'].create({'user_id': self.env.uid})
+        Timesheet = self.env['account.analytic.line']
+        Task = self.env['project.task']
+        today = Date.context_today(self.env.user)
+
+        task = Task.with_context(default_project_id=self.project_template.id).create({
+            'name': 'first task',
+            'partner_id': self.partner_b.id,
+            'allocated_hours': 48,
+            'sale_line_id': self.so.order_line[0].id
+        })
+
+        timesheet = Timesheet.create({
+            'project_id': self.project_template.id,
+            'task_id': task.id,
+            'name': 'my first timesheet',
+            'unit_amount': 30,
+        })
+        timesheet.action_timer_start()
+        timesheet.user_timer_id.timer_start = today - timedelta(days=1)
+
+        context = {
+            'active_model': 'sale.order',
+            'active_ids': [self.so.id],
+            'active_id': self.so.id,
+            'default_journal_id': self.company_data['default_journal_sale'].id
+        }
+
+        wizard = self.env['sale.advance.payment.inv'].with_context(context).create({
+            'advance_payment_method': 'delivered',
+        })
+        wizard.create_invoices()
```

Comment:
To be sure the compute works as expected and we don't stop any running timers when `advance_payment_method` is different than `delivered`. 🙂 

```suggestion
        wizard = self.env['sale.advance.payment.inv'].with_context(context).create({
            'advance_payment_method': 'percentage',
        })
        self.assertTrue(wizard.has_timer_running)
        wizard.create_invoices()
        self.assertTrue(timesheet.is_timer_running, 'The running timer should still be running since we did not create a regular invoice.')
        wizard.advance_payment_method = 'delivered'
        wizard.create_invoices()
        self.assertFalse(timesheet.is_timer_running, 'The running timer should be stopped since we generate a regular invoice.')
```

Note: I did not test my suggestion